src/formatDate.js

```diff
@@ -9,6 +9,62 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const dateObj = parseData(date, fromFormat);
+
+  return getData(dateObj, toFormat);
 }
 
+function parseData(date, fromFormat) {
+  const dateObj = {};
+
+  for (let i = 0; i < 3; i++) {
+    const part = date.split(fromFormat[3])[i];
+
+    if (fromFormat[i] === 'DD') {
+      dateObj.day = part;
+    }
+
+    if (fromFormat[i] === 'MM') {
+      dateObj.month = part;
+    }
+
+    if (fromFormat[i] === 'YYYY') {
+      dateObj.year = part;
+    }
+
+    if (fromFormat[i] === 'YY') {
+      if (part < 30) {
```

Comment:
You should ensure that the year part is exactly two digits. If the year part is a single digit, this code will yield incorrect results. For instance, '5' would become '205' instead of '2005'.